core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -72,4 +74,87 @@ class SparkContextSuite extends FunSuite with LocalSparkContext {
     val byteArray2 = converter.convert(bytesWritable)
     assert(byteArray2.length === 0)
   }
+
+  test("addFile works") {
+    val file = new File("somefile")
+    val absolutePath = file.getAbsolutePath
+    try {
+      val pw = new PrintWriter(file)
+      pw.print("somewords")
+      pw.close()
+      val length = file.length()
+      sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+      sc.addFile(file.getAbsolutePath)
+      sc.parallelize(Array(1), 1).map(x => {
+        val gotten = new File(SparkFiles.get(file.getName))
```

Comment:
I guess using `local` doesn't incur serialization? Because this would fail to serialize `file` in a real cluster.